src/components/tracksList.vue

```diff
@@ -0,0 +1,46 @@
+<script lang="ts" setup>
+import filters from "@/utils/filters";
+import { useAuth0 } from "@auth0/auth0-vue";
+import { onMounted, ref } from "vue";
+
+// import ProtectedImage from "./ProtectedImage.vue";
+
+const props = defineProps<{
+  src: any;
+  id: any;
+  title: any;
+  album: any;
+  subtype: any;
+}>();
+
+const source = ref("");
+const trackNumber = ref();
+const trackTitle = ref("");
+const trackAlbum = ref("");
+const trackSubtype = ref("");
+
+const { getAccessTokenSilently } = useAuth0();
+
+onMounted(async () => {
+  const token = await getAccessTokenSilently();
+
+  source.value = filters.authorizedUrl(props.src, token);
+  trackNumber.value = filters.authorizedUrl(props.id, token);
```

Comment:
What is the reason for using the `authorizedUrl` filter here? :)

Reply:
I said to myself that an authorization was needed for each piece of data that we want to have. not true ? If I did not write well, can you suggest me something or a correction?

Reply:
Its a good thought! I think, however, that the `authorizedUrl` util is primarily used to authorize image urls (or URLs in general).

If you look at the code for the `authorizedUrl` util, you'll see it is setting some headers on a `URL` object.
We probably don't need to pass numbers through, as numbers aren't URLs.

Reply:
Ok ! Thanks - How do you suggest me to do then @sifferhans ?

Reply:
I think you can simply use the props directly instead of using separate `ref`s.

You can also just use the `ProtectedImage` component instead of an `img` element, which will handle authorization of the image source for you 😊

Reply:
Okay ! I will try it afterwards !

Reply:
It's actually very rare you'd need the function `authorizedUrl` in your code. Maybe a good advice is to first try a minimalistic approach (less code) and add what's needed - as you understand their usage.